src/index.ts

```diff
@@ -227,63 +226,56 @@ const main = async () => {
     },
   );
 
-  tool('list_problems', 'List all problems known on Dynatrace', {}, async ({}) => {
-    const dtClient = await createDtHttpClient(
-      dtEnvironment,
-      scopesBase.concat('environment-api:problems:read'),
-      oauthClientId,
-      oauthClientSecret,
-      dtPlatformToken,
-    );
-    const result = await listProblems(dtClient);
-    if (!result || result.length === 0) {
-      return 'No problems found';
-    }
-    return `Found these problems: ${result.join(',')}`;
-  });
-
   tool(
-    'get_problem_details',
-    'Get details of a problem on Dynatrace',
+    'list_problems',
+    'List all problems (dt.davis.problems) known on Dynatrace, sorted by their recency. An additional filter can be provided using DQL filter.',
     {
-      problemId: z.string().optional(),
+      additionalFilter: z
+        .string()
+        .optional()
+        .describe(
+          'Additional filter for DQL statement for dt.davis.problems, e.g., \'entity_tags == array("dt.owner:team-foobar", "tag:tag")\'',
+        ),
+      maxProblemsToDisplay: z.number().default(10).describe('Maximum number of problems to display in the response.'),
     },
-    async ({ problemId }) => {
+    async ({ additionalFilter, maxProblemsToDisplay }) => {
       const dtClient = await createDtHttpClient(
         dtEnvironment,
-        scopesBase.concat('environment-api:problems:read'),
+        scopesBase.concat('storage:events:read', 'storage:buckets:read'),
         oauthClientId,
         oauthClientSecret,
         dtPlatformToken,
       );
-      const result = await getProblemDetails(dtClient, problemId);
-
-      let resp =
-        `The problem ${result.displayId} with the title ${result.title} (ID: ${result.problemId}).` +
-        `The severity is ${result.severityLevel}, and it affects ${result.affectedEntities.length} entities:`;
-
-      for (const entity of result.affectedEntities) {
-        resp += `\n- ${entity.name} (please refer to this entity with \`entityId\` ${entity.entityId?.id})`;
-      }
-
-      resp += `The problem first appeared at ${result.startTime}\n`;
-      if (result.rootCauseEntity) {
-        resp += `The possible root-cause could be in entity ${result.rootCauseEntity?.name} with \`entityId\` ${result.rootCauseEntity?.entityId?.id}.\n`;
-      }
-
-      if (result.impactAnalysis) {
-        let estimatedAffectedUsers = 0;
-
-        result.impactAnalysis.impacts.forEach((impact) => {
-          estimatedAffectedUsers += impact.estimatedAffectedUsers;
+      // get problems (uses fetch )
```

Comment:
Sorry, I am wearing my nitpicking pants today 😅

```suggestion
      // get problems (uses fetch)
```